lib/infra/infra-stack.ts

```diff
@@ -793,6 +811,47 @@ export class InfraStack extends Stack {
     }
   }
 
+  private applyFeatureAdjustments(): void {
```

Comment:
lets remove these functions, lets handle the feature related configuration logic in https://github.com/opensearch-project/opensearch-cluster-cdk/blob/main/lib/infra/infra-stack.ts#L796. 

As of now we only need to handle plugin installation, and lets use the new heap override param to handle heap requirements.